sql/planbuilder/from.go

```diff
@@ -314,11 +314,10 @@ func columnsToStrings(cols ast.Columns) []string {
 }
 
 func (b *Builder) resolveTable(tab, db string, asOf interface{}) *plan.ResolvedTable {
-	table, _, err := b.cat.TableAsOf(b.ctx, db, tab, asOf)
-	if err != nil {
-		b.handleErr(err)
+	table, database, err := b.cat.TableAsOf(b.ctx, db, tab, asOf)
+	if sql.ErrAsOfNotSupported.Is(err) {
+		table, database, err = b.cat.Table(b.ctx, db, tab)
```

Comment:
If `asOf` is not nil, should we still throw an error here so that we don't drop a requested `asOf` clause?